addons/binding/org.openhab.binding.spotify/src/main/java/org/openhab/binding/spotify/internal/api/SpotifyApi.java

```diff
@@ -0,0 +1,293 @@
+    }
+
+    /**
+     * Calls the Spotify player Web Api with the given method and appends the given url as parameters of the call to
+     * Spotify.
+     *
+     * @param method Http method to perform
+     * @param url url path to call to spotify
+     * @return the response give by Spotify
+     */
+    private ContentResponse requestPlayer(HttpMethod method, String url) {
+        return requestPlayer(method, url, "");
+    }
+
+    /**
+     * Calls the Spotify player Web Api with the given method and appends the given url as parameters of the call to
+     * Spotify.
+     *
+     * @param method Http method to perform
+     * @param url url path to call to spotify
+     * @param requestData data to pass along with the call as content
+     * @return the response give by Spotify
+     */
+    private ContentResponse requestPlayer(HttpMethod method, String url, String requestData) {
+        return request(method, SPOTIFY_API_PLAYER_URL + (url.isEmpty() ? "" : ('/' + url)), requestData);
+    }
+
+    /**
+     * Calls the Spotify Web Api with the given method and given url as parameters of the call to Spotify.
+     *
+     * @param method Http method to perform
+     * @param url url path to call to spotify
+     * @param requestData data to pass along with the call as content
+     * @return the response give by Spotify
+     */
+    private ContentResponse request(HttpMethod method, String url, String requestData) {
+        logger.debug("Request: ({}) {} - {}", method, url, requestData);
+        final Function<HttpClient, Request> call = httpClient -> httpClient.newRequest(url).method(method)
+                .header("Accept", CONTENT_TYPE).content(new StringContentProvider(requestData), CONTENT_TYPE);
+        try {
+            try {
+                final AccessTokenResponse accessTokenResponse = oAuthClientService.getAccessTokenResponse();
+                final String accessToken = accessTokenResponse == null ? null : accessTokenResponse.getAccessToken();
+
+                if (accessToken == null || accessToken.isEmpty()) {
+                    throw new SpotifyAuthorizationException("No spotify accesstoken. Is this thing authorized?");
+                } else {
+                    return connector.request(call, BEARER + accessToken);
+                }
+            } catch (SpotifyTokenExpiredException e) {
+                // Retry with new access token
+                return connector.request(call, BEARER + oAuthClientService.refreshToken().getAccessToken());
+            }
+        } catch (IOException e) {
+            logger.debug("Request failed to during refresh token: ", e);
+            throw new SpotifyException(e.getMessage());
+        } catch (OAuthException | OAuthResponseException e) {
+            logger.debug("Request authorization failed to during refresh token: ", e);
+            throw new SpotifyAuthorizationException(e.getMessage());
```

Comment:
Consider to not discard the the stacktrace here

Reply:
As far as I know its best practice to either log or throw, but not you are kinda doing both, I think that passing the exception as a cause to the new one should eliminate the logging here